### PR TITLE
Cypress: Add Volume cases

### PR DESCRIPTION
**Pull request by DaiYuzeng**
#### Releated cases:
- https://harvester.github.io/tests/manual/volumes/create-image-from-volume/
- https://harvester.github.io/tests/manual/volumes/create-volume-root-disk-vm-image-form/
- https://harvester.github.io/tests/manual/volumes/create-volume-root-disk-vm-image-form-label/
- https://harvester.github.io/tests/manual/volumes/delete-volume-that-is-not-attached-to-vm/
- https://harvester.github.io/tests/manual/volumes/delete-volume-that-was-attached-to-vm-but-is-not-now/

### CODE REVIEW

File: cypress/testcases/volumes.spec.ts [hunk 1]
```diff
@@ -0,0 +1,202 @@
...
+ * 2. Delete the VM but not the volume
+ * 3. Verify Volume still exists
+ * 4. Delete the volume
+ * Expected Results
+ * 1. VM should create
+ * 2. VM should delete
+ * 3. Volume should still show in Volume list
+ * 4. Volume should delete
+ */
+ describe.only('Delete volume that was attached to VM but now is not', () => {
```
Comment (WuJun2016) [inline cypress/testcases/volumes.spec.ts:118]: delete '.only'

Comment (DaiYuzeng) [reply cypress/testcases/volumes.spec.ts]: done.

**Comment by noahgildersleeve** [comment]
@DaiYuzeng This is marked as a draft PR. Is it ready for merging?

**Comment by DaiYuzeng** [comment]
@noahgildersleeve Please take a look, thanks!

**Review by TachunLin — CHANGES_REQUESTED** [review]
Verified running volume.spec.ts test suite, some of the cases failed
Not sure it's related to network latency or dependency from other test

- [ ]  ***FAILED***  Create image from Volume
- [x]  ***PASS*** Create volume root disk VM Image Form
- [ ]  ***FAILED***  Delete volume that was attached to VM but now is not

image

* Create image from Volume failed in 
Found failure originate in 
[pageobjects/image.po.ts:128:43](https://192.168.0.131/__/#)
image

* `Delete volume that was attached to VM but now is not` failed in 
[pageobjects/volume.po.ts:90:12](https://192.168.0.131/__/#)
image

**Comment by n313893254** [comment]
Hi, @DaiYuzeng. The VM [PR](https://github.com/harvester/tests/pull/359) has merged, Please let me know if it caused any issues.

**Comment by DaiYuzeng** [comment]
> Hi, @DaiYuzeng. The VM [PR](https://github.com/harvester/tests/pull/359) has merged, Please let me know if it caused any issues.

Thanks for the changes!
I am going to resolve the conflicts and then do some tests.
But I am working on an issue of Harvester Node Driver, I'll work on this issue 2/3 days later.

File: cypress/constants/constants.ts [hunk 2]
```diff
@@ -5,13 +5,13 @@ export class Constants {
...
-    public dashboardUrl = '/c/local/harvester/harvesterhci.io.dashboard';
-    public settingsUrl = '/c/local/harvester/harvesterhci.io.setting';
-    public uiSourceUrl = '/c/local/harvester/harvesterhci.io.setting/ui-source?mode=edit';
-    public hostsPage = '/c/local/harvester/harvesterhci.io.host';
-    public supportPage = '/c/local/harvester/support';
-    public vmPage = '/c/local/harvester/kubevirt.io.virtualmachine';
-    public settingBaseUrl = '/c/local/harvester/harvesterhci.io.setting';
-    public volumePage = '/c/local/harvester/harvesterhci.io.volume';
-    public imagePage = '/c/local/harvester/harvesterhci.io.virtualmachineimage';
+    public dashboardUrl = '/harvester/c/local/harvesterhci.io.dashboard';
```
Comment (DaiYuzeng) [inline cypress/constants/constants.ts:8]: @WuJun2016 @n313893254 Please take a glimpse.

File: cypress/cypress.env.json.example [hunk 3]
```diff
@@ -14,9 +14,12 @@
       }
     ]
   },
-  "image":
-  {
+  "largeImage": {
```
Comment (DaiYuzeng) [inline cypress/cypress.env.json.example:17]: And here

File: cypress/pageobjects/virtualmachine.po.ts [hunk 4]
```diff
@@ -187,6 +179,22 @@ export class VmsPage extends CruResourcePo {
     cy.get('.cru-resource-footer').contains('Save').click()
   }
 
+  public delete(namespace:string, name: string, { removeRootDisk }: { removeRootDisk?: boolean } = { removeRootDisk: true }) {
```
Comment (DaiYuzeng) [inline cypress/pageobjects/virtualmachine.po.ts:182]: @WuJun2016 @n313893254 Please take a glimpse.

File: cypress/utils/components/checkbox.po.ts [hunk 5]
```diff
@@ -6,8 +6,8 @@ export default class CheckboxPo extends ComponentPo {
       return
     }
 
-    this.self().find('[role="checkbox"]').invoke('attr', 'aria-checked').then(isChecked => {
-      if ((newValue && !isChecked) || (!newValue && isChecked)) {
+    this.self().find('input').invoke('attr', 'value').then(isChecked => {
```
Comment (DaiYuzeng) [inline cypress/utils/components/checkbox.po.ts:9]: It seems like the selector `[role="checkbox"]` doesn't work.

File: cypress/utils/components/cru-resource.po.ts [hunk 6]
```diff
@@ -14,12 +14,11 @@ export default class CruResourcePo extends PagePo {
   public type = '';
   public realType = '';
   public storeType: string|undefined = undefined;
-  
-  private footerButtons = '.cru-resource-footer'
-  private confirmRemove = '.card-container.prompt-remove'
-  private searchInput = '.search'
-  private actionMenu = '.list-unstyled.menu'
-  private actionMenuIcon = '.icon-actions'
+  public footerButtons = '.cru-resource-footer'
```
Comment (DaiYuzeng) [inline cypress/utils/components/cru-resource.po.ts:17]: Refator these members from private to public

**Review by WuJun2016 — CHANGES_REQUESTED** [review]

**Comment by DaiYuzeng** [comment]
The PR is ready for review, please take a look, thanks!

**Comment by DaiYuzeng** [comment]
@lanfon72 @TachunLin Hi! This PR is ready for review, please take a look, thanks!
The last case in image, the last case in volume and the first case in host can't be past, we'll fix these cases at incoming commit.

**Comment by TachunLin** [comment]
* Tested can finish all **Image** test cases on Harvester `v1.1.0`, it takes 6 minutes 13 seconds execution time
image

* Tested can finish all **Volume** test cases on Harvester `v1.1.0`, it takes 3 minutes 15 seconds execution time
image

* Tested can finish all **Network** test cases on Harvester `v1.1.0`, it takes 1 minutes 6 seconds execution time
image

**Comment by TachunLin** [comment]
1. To make the Image/`Create a ISO image via upload` works, we may need to change `/cypress/cypress/fixtures/cirros-0.5.2-aarch64-disk.img` to executable 
image

2. We can use minio to setup as NFS backup target for the settings test cases

**Comment by DaiYuzeng** [comment]
> While running the `hosts.spec.ts`, the 3rd test plan `Check Add disk` would failed at Line 70 `testcases/dashboard/hosts.spec.ts:70:8`
> 
> image

Hi @TachunLin ! We would like to file a new [issue](https://github.com/harvester/tests/issues/599) to track this failed case. Thanks!

**Review by TachunLin — APPROVED** [review]
According to above test result. 
Checked the related test cases can run correctly on Harvester v1.1.0 
For the hosts.spec.ts related issue would be track separately at test issue https://github.com/harvester/tests/issues/599

**Comment by DaiYuzeng** [comment]
Hi @TachunLin ! The newest commit has been merged into this PR. It will affect two cases:

1. network
2. virtualmachine/network

image

Please take a look! Thanks! 

cc @n313893254 @WuJun2016

**Review by TachunLin — CHANGES_REQUESTED** [review]
@n313893254, @DaiYuzeng 
When tested the network/network.spec.ts, the Create Vlan1 and Vlan2 cases encounter failure. 
It would probably related to the column id changes. Please help to take a look. Thanks

* Create Vlan1 failed 
image

* Create Vlan2 failed
image

**Comment by TachunLin** [comment]
Checked can pass the `Create Vlan1` and failed the `Create Vlan2`, it's correct behavior since Vlan 2 can routed on local kvm environment.  
image

Can pass all `virtualmachine/network` test cases
image

**Comment by noahgildersleeve** [comment]
@WuJun2016 Could you check if your changes are still needed/fixed when you get a chance so we can get this merged?

**Comment by DaiYuzeng** [comment]
Hi @noahgildersleeve @WuJun2016 ! I think this PR has been ready to merge, thank you!

**Comment by WuJun2016** [comment]
> Checked can pass the `Create Vlan1` and failed the `Create Vlan2`, it's correct behavior since Vlan 2 can't be routed on local kvm environment.

@noahgildersleeve   I think it can be merged. If there is any problem, we can submit a new pr to solve it.

**Review by WuJun2016 — APPROVED** [review]
LGTM, thanks

**Review by n313893254 — APPROVED** [review]

**Pull request merged**